app/controllers/friendly_url_redirect_controller.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+// Node.js core dependencies
+const logger = require('winston')
+
+// Custom dependencies
+const productsClient = require('../services/clients/products_client')
+const response = require('../utils/response')
+const errorResponse = response.renderErrorView
+
+// Constants
+const messages = {
+  internalError: 'We are unable to process your request at this time'
+}
+
+module.exports = (req, res) => {
+  const {serviceNamePath, productNamePath} = req.params
+  productsClient.product.getByProductPath(serviceNamePath, productNamePath)
+    .then(product => {
+      logger.info(`Redirecting to ${product.links.pay.href}`)
+      return res.redirect(product.links.pay.href)
+    })
+    .catch(err => {
+      return errorResponse(req, res, messages.internalError, err.errorCode || 500)
```

Comment:
Perhaps handle 404 and 500 differently (showing the user what happened) here is better than always 500.

Reply:
This is as expected. The error code would indeed by a 404 in this case. I also spoke to Sonia whio is happy with the error message we have here for now.